agent/app/agent_compatibility_unpecified.go

```diff
@@ -0,0 +1,23 @@
+// +build !linux
+// Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package app
+
+import (
+	"github.com/aws/amazon-ecs-agent/agent/engine"
+)
+
+func (agent *ecsAgent) checkCompatibility(engine engine.TaskEngine) error {
```

Comment:
When exactly will this be called? 
also nit - I guess the file name is `agent_compatibility_unspecified.go`?

Reply:
This still gets called when we reload the state file in the app startup. On Windows / OS X, it won't do anything.

Also thanks! Spelling is hard.

Reply:
aah ok, used to seeing `_windows` everywhere, `unspecified` was new. So just confirmed.

Reply:
The task cgroup checks need happen only on supported platforms`(linux)` but we still need to maintain compatibility on `windows`